src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java

```diff
@@ -78,18 +76,27 @@ public void distribute() {
   }
 
   private void processJob(CollectionExerciseJob job) {
-    List<SampleUnit> mappedSampleUnits =
-        getMappedSampleUnits(job.getSampleSummaryId(), job.getCollectionExerciseId().toString());
-
-    boolean hasErrors = false;
-
-    for (SampleUnit msu : mappedSampleUnits) {
-      try {
-        sampleUnitSender.sendSampleUnit(msu);
-      } catch (CTPException e) {
-        hasErrors = true;
-        log.with("sample_unit_id", msu.getId())
-            .error("Failed to send a sample unit to queue and update state with ID", e);
+    SampleSummary sampleSummary = sampleSummaryRepository.findById(job.getSampleSummaryId());
+
+    hasErrors = false;
+
+    if (sampleSummary.getState() == SampleState.ACTIVE) {
```

Comment:
Did you not find the other way easier to read/follow?

Reply:
The way it was implemented before was reading the whole sample into memory. This way it'll only ever have one sample unit in memory, which is safer.